### PR TITLE
Move to Quarkus 2.0.2.Final

### PR DESCRIPTION
**Pull request by rsvoboda**
Move to Quarkus 2.0.2.Final

Reworks and enhances https://github.com/quarkus-qe/quarkus-startstop/pull/92 as @kshpak is on PTO

### CODE REVIEW

File: testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java [hunk 1]
```diff
@@ -200,7 +202,9 @@ public void supportedExtensionsSubsetB(TestInfo testInfo) throws Exception {
 
     @Test
     public void supportedExtensionsSubsetC(TestInfo testInfo) throws Exception {
-        testRuntime(testInfo, supportedEx.get(2), MvnCmds.MVNW_DEV);
+        List<CodeQuarkusExtensions> supportedExtensionsSubsetC = supportedEx.get(2);
+        supportedExtensionsSubsetC.add(CodeQuarkusExtensions.QUARKUS_RESTEASY);
+        testRuntime(testInfo, supportedExtensionsSubsetC, MvnCmds.MVNW_DEV);
```
Comment (Sgitario) [inline testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java:207]: I miss the point here. I guess what we want is to verify the third group of supported extensions, but ensuring we have the resteasy extension which contains the code starter, but I missed why we want to verify this use case as the behaviour of the old test case has changed.
If this use case is relevant, I would add a new test case and keep the `supportedExtensionsSubsetC` as it was, but maybe a comment to clarify this would help too.

**Review by Sgitario — COMMENTED** [review]
Lgtm, just minor clarifications needed from my side.

File: app-full-microprofile/src/main/resources/application.properties [hunk 2]
```diff
@@ -10,9 +10,9 @@ mp.jwt.verify.publickey.location=META-INF/resources/publicKey.pem
...
 quarkus.smallrye-jwt.enabled=true
 
-quarkus.jaeger.service-name=Demo-Service-A
-quarkus.jaeger.sampler-type=const
-quarkus.jaeger.sampler-param=1
-quarkus.jaeger.endpoint=http://localhost:14268/api/traces
+#quarkus.jaeger.service-name=Demo-Service-A //TODO Watch https://github.com/quarkusio/quarkus/issues/18131
+#quarkus.jaeger.sampler-type=const
+#quarkus.jaeger.sampler-param=1
+#quarkus.jaeger.endpoint=http://localhost:14268/api/traces
```
[the remark this answers is not in the document]
Comment (rsvoboda) [reply app-full-microprofile/src/main/resources/application.properties:16]: That's coming from Kyrylo's PR, I can double check this and find the minimal change

File: testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java (hunk 1 above)
Comment (rsvoboda) [reply testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java:207]: That's coming from Kyrylo's PR, I will add comment to explain the need for this.
tl;dr: resteasy or spring-web extension is needed to provide index.html file we are checking in the tests.

Comment (Sgitario) [reply testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java:207]: Then, all the combinations must contain either the resteasy or spring-web? Otherwise, I will add a comment to clarify this for others in the future too.

Comment (rsvoboda) [reply testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java:209]: subsetA and subsetB contain the needed extensions, so just subsetC was needed to be enhanced "manually"

**Review by Sgitario — APPROVED** [review]
lgtm, will merge it once CI is green (but the code quarkus job).
